.gitignore

```diff
@@ -1,13 +1,39 @@
-*.log
-*.tlog
-*.lastbuildstate
-Debug/
-*.obj
-*.VC.db
-*.VC.opendb
-*.suo
-*.aps
-ipch/
-*.user
-SharkCage/.vs/
-SharkCage/post_build_output
+# Ignore everything
+*
+
+# Whitelist subdirectories
+!*/
+
+# general
+!*.gitignore
+!*.yml
+!*.md
+!*.txt
+!*.pdf
+!*.docx
+!*.bmp
+
+# project related
+!*.sln
+!*.vcxproj
+!*.vcxproj.filters
+!*.csproj
+!*.config
+!*.settings
+!*.ps1
+
+# c++ / c#
+!*.h
+!*.cpp
+!*.rc
+!*.cs
+!*.resx
+
+# special rules for built sources
```

Comment:
Still getting some unwanted files in build-folders to show up as untracked:
![image](https://user-images.githubusercontent.com/16246147/40540659-32bd31d0-6019-11e8-8bce-635ecb9ae0e3.png)

How about just explicitely ignoring everything in those folders again?
```
# special rules for built sources
**/Debug/**/*.*
**/Release/**/*.*
**/bin/**/*.*
**/obj/**/*.*
```